### PR TITLE
Implement Histograms

### LINKED ISSUE #5: Histogram support
Prometheus now has histograms to complement summaries:
- https://prometheus.io/docs/concepts/metric_types/#histogram
- https://prometheus.io/docs/practices/histograms/

It'd be great to be able to export them from Haskell code.

### PR DESCRIPTION
**Pull request by jml**
Fixes #5 

See
* https://prometheus.io/docs/concepts/metric_types/#histogram
* https://prometheus.io/docs/practices/histograms/

for more information.

My Emacs really dislikes the indentation used in prometheus-haskell, so I may have misformatted some things.

Happy to change things as needed.

### CODE REVIEW

File: prometheus-client/src/Prometheus/Metric/Observer.hs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    observe :: MonadMonitor m => Double -> Metric metric -> m ()
+
+-- | Adds the duration in seconds of an IO action as an observation to an
+-- observer metric.
+observeDuration :: Observer metric => IO a -> Metric metric -> IO a
+observeDuration io metric = do
+    start  <- getCurrentTime
+    result <- io
+    end    <- getCurrentTime
+    observe (fromRational $ toRational $ end `diffUTCTime` start) metric
```
Comment (brian-brazil) [inline prometheus-client/src/Prometheus/Metric/Observer.hs]: Watch out for time going backwards, a minimum of 0 would  be wise.

Comment (jml) [inline prometheus-client/src/Prometheus/Metric/Observer.hs]: Good point. https://github.com/fimad/prometheus-haskell/issues/9 points to using different routines, which (I think—haven't actually dug into it) provide better guarantees. My plan was to address that in a separate PR, but if @fimad is OK with it I can do it in this one.

Comment (fimad) [inline prometheus-client/src/Prometheus/Metric/Observer.hs]: Let's do separate PRs, I generally prefer each PR have a single topic.

**Comment by fimad** [comment]
Would you like a new release with this change, or should I wait for a follow up for #9?

**Comment by jml** [comment]
@fimad Thanks for getting to this. I reckon wait for a follow-up—I ought to be able to get to it this week.
